handlers/search.go

```diff
@@ -69,7 +69,9 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	searchType := vars["type"]
 	search := vars["search"]
-	search = strings.Replace(search, "0x", "", -1)
+	if !utils.IsValidEnsDomain(search) {
+		search = strings.Replace(search, "0x", "", -1)
+	}
```

Comment:
While this fixes the problem for ENS, it does not fix the same exact problem for, for example, slots, epochs and likely validator names:
Searching for "0x100" will show you both slot and epoch 100 (and more wrong results).

I think this `strings.Replace` should only be part of `case`s that work with something that might have an optional "0x" prefix like addresses or validator public keys.

Weirdly enough, this is already done for transactions, see https://github.com/gobitfly/eth2-beaconchain-explorer/blob/dbea213b5f11f852494690aeaecf42a4673c020c/handlers/search.go#L138

Reply:
Refactored the whole thing a bit ... pls have a look ;-)